stl/inc/ranges

```diff
@@ -3427,6 +3431,489 @@ namespace ranges {
+                noexcept(static_cast<decltype(_Val)>(*_It)))
+                : _Val(*_It) {}
+
+            remove_cv_t<range_reference_t<_Vw>> _Val;
+        };
+
+    protected:
+        /* [[no_unique_address]] */ _Non_propagating_cache<_Cache_wrapper, false> _Inner{};
+    };
+
+    template <class _Vw, class _Pat>
+        requires is_reference_v<range_reference_t<_Vw>>
+    class _Join_with_view_base<_Vw, _Pat> : public view_interface<join_with_view<_Vw, _Pat>> {
+    };
+
+    template <input_range _Vw, forward_range _Pat>
+        requires view<_Vw> && input_range<range_reference_t<_Vw>> && view<_Pat> //
+            && _Compatible_joinable_ranges<range_reference_t<_Vw>, _Pat>
+    class join_with_view : public _Join_with_view_base<_Vw, _Pat> {
+    private:
+        template <bool _Const>
+        using _InnerRng = range_reference_t<_Maybe_const<_Const, _Vw>>;
+
+        /* [[no_unique_address]] */ _Vw _Range{};
+        /* [[no_unique_address]] */ _Pat _Pattern{};
+
+        template <bool _Const>
+        class _Sentinel;
+
+        template <bool _Const>
+        struct _Category_base {};
+
+        template <bool _Const>
+            requires forward_range<_Maybe_const<_Const, _Vw>> && forward_range<_InnerRng<_Const>> //
+                && is_reference_v<_InnerRng<_Const>>
+        struct _Category_base<_Const> {
+            using _Outer       = _Maybe_const<_Const, _Vw>;
+            using _Inner       = _InnerRng<_Const>;
+            using _PatternBase = _Maybe_const<_Const, _Pat>;
+
+            using iterator_category = conditional_t<
+                !is_lvalue_reference_v<common_reference_t<range_reference_t<_Inner>, range_reference_t<_PatternBase>>>,
+                input_iterator_tag,
+                conditional_t<common_range<_Inner> && common_range<_PatternBase> //
+                                  && derived_from<_Iter_cat_t<iterator_t<_Outer>>, bidirectional_iterator_tag> //
+                                  && derived_from<_Iter_cat_t<iterator_t<_Inner>>, bidirectional_iterator_tag> //
+                                  && derived_from<_Iter_cat_t<iterator_t<_PatternBase>>, bidirectional_iterator_tag>,
+                    bidirectional_iterator_tag,
+                    conditional_t<derived_from<_Iter_cat_t<iterator_t<_Outer>>, forward_iterator_tag> //
+                                      && derived_from<_Iter_cat_t<iterator_t<_Inner>>, forward_iterator_tag> //
+                                      && derived_from<_Iter_cat_t<iterator_t<_PatternBase>>, forward_iterator_tag>,
+                        forward_iterator_tag, input_iterator_tag>>>;
+        };
+
+        template <bool _Const>
+        class _Iterator : public _Category_base<_Const> {
+        private:
+            friend join_with_view;
+
+            template <bool>
+            friend class _Iterator;
+
+            template <bool>
+            friend class _Sentinel;

```

Comment:
After `friend join_with_view;`, these are unnecessary. `_Iterator` and `_Sentinel` can access anything that the parent can access, WG21-N4910 [class.access.nest]/1 "A nested class is a member and as such has the same access rights as any other member." Thus, granting friendship to the parent effectively grants friendship to its nested classes.